### ISSUE TITLE
[documentation] Keep a changelog as we make changes to the code

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1103]
I believe @dbeltrankyl writes a changelog for some releases, but perhaps we could add a changelog entry for each merge request. This is common practice in many other projects, and having the changelog always up to date gives users a better idea of what to expect for our next release.

I think we had discussed it before, but it came up again in a startR meeting where @V-Agudetse asked if we had a changelog.

Now that we have startR users, LanguageTech, EDITO, DestinE, Kit, and other external users, it might be a good idea to introduce this.

We can use a merge request template to remind Dani, I, other contributors (which we are also getting more now! :tada: , not only new users), that we need to update it. Then when the release time arrives, it should be easier to produce & send the changelog to internal users.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 7, 2023, 11:11

I've been adding the summarized changelog notes in the tag release for a while, but we don't have a file. 

For example, the 4.0.22

```
 v4.0.0b22
 13bf2b43 · fix wrappers · 2 weeks ago
- Parameters fixes
- testcase fix 
- updated config parser version ( added ^, for fixed)
- wrapper fix
- sorted load
```

**Comment by kinow** [comment]
That's what I replied to @V-Agudetse :slight_smile: But I think it would be better for users (especially our new external users) to know what we are currently working as well, so they can prepare themselves for the next version, or even send issues in case they identify something that may break their environments.

**Comment by kinow** [comment]
mentioned in issue digital-twins/de_340/project_management#179

**Comment by kinow** [comment]
Some days ago @mcastril was asked about the number of features added in Autosubmit in 2022 and 2023. This was very hard to get from reading the existing comments in the tags, or our current changelog file.

This needs to be improved so that it is easier to both ourselves and external users to track our changes in each release. There are beta versions released too, where would be good to keep track of what's in each one, and clearly state if they are safe to use in production or not. We could even think about a LTS version for the future.

The ClimateDT workflow has a changelog template similar to the one from AQUA and Cylc, where you have sections for features, bugs, breaking changes, etc. (the ClimateDT was decided by @Lerriola / @ainagaya / @franra9, and with that one we could just count the number of lines in each "Added" section to know how many features were added, for instance).

**Comment by kinow** [comment]
We have a changelog now, so closing this. Not sure when this was implemented exactly, but I guess we can leave 4.1 as the milestone.
